tests/ext-popup.test.js

```diff
@@ -23,187 +26,205 @@ function observeChange(element) {
   });
 }
 
-test('start monitoring all extensions from the popup', async () => {
+const getMonitorStatusSpyFn = jest.spyOn(ExtListen, 'getMonitorStatus');
+const startMonitorSpyFn = jest.spyOn(ExtListen, 'startMonitor');
+const stopMonitorSpyFn = jest.spyOn(ExtListen, 'stopMonitor');
+
+afterEach(() => {
+  getMonitorStatusSpyFn.mockReset();
+  startMonitorSpyFn.mockReset();
+  stopMonitorSpyFn.mockReset();
+});
+
+test('clicking on start monitoring button should display that the extensions are being monitored', async () => {
   document.body.innerHTML = popupBody;
 
-  const getMonitorStatusFn = jest.spyOn(ExtListen, 'getMonitorStatus');
-  const startMonitorFn = jest.spyOn(ExtListen, 'startMonitor');
+  // no extensions are being monitored initially
+  getMonitorStatusSpyFn.mockResolvedValueOnce(false);
+  // extensions are being monitored after the start monitoring button is clicked
+  getMonitorStatusSpyFn.mockResolvedValueOnce(true);
+  startMonitorSpyFn.mockResolvedValue(true);
 
   const popup = new Popup();
+  await popup.init();
 
   const startBtn = popup.startMonitorAllBtn;
   const stopBtn = popup.stopMonitorAllBtn;
   const monitorStatus = popup.monitorStatusText;
   const errorText = popup.errorMsgText;
 
-  getMonitorStatusFn.mockResolvedValueOnce(false);
-  getMonitorStatusFn.mockResolvedValueOnce(true);
-  startMonitorFn.mockResolvedValue(true);
-
-  await popup.init();
-
-  // popup at initial - no extensions being watched
-  expect(startBtn.hasAttribute('disabled')).toBeFalsy();
-  expect(stopBtn.hasAttribute('disabled')).toBeTruthy();
-
-  expect(monitorStatus.textContent).toBe('No extensions are being monitored');
-  expect(monitorStatus.classList.contains('failure')).toBeTruthy();
-  expect(monitorStatus.classList.contains('success')).toBeFalsy();
-  expect(errorText.textContent).toBe('');
```

Comment:
The initial UI behaviour of the browserAction popup is not necessary to be checked for the purpose of the test - `clicking on start monitoring button should display that the extensions are being monitored`.

Rather, I have added - 
```js
  // no extensions are being monitored initially
  getMonitorStatusSpyFn.mockResolvedValueOnce(false);
```
above which explains the initial state of the browserAction popup.
I think this improves the test's quality.